subsys/net/lib/websocket/websocket.c

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_DEBUG_WEBSOCKET)
+#define SYS_LOG_DOMAIN "ws"
+#define NET_SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <zephyr.h>
+#include <string.h>
+#include <strings.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <version.h>
+
+#include <net/net_ip.h>
+#include <net/websocket.h>
+
+#include <mbedtls/base64.h>
+#include <mbedtls/sha1.h>
+
+#define BUF_ALLOC_TIMEOUT 100
+
+#define HTTP_CRLF "\r\n"
+
+/* From RFC 6455 chapter 4.2.2 */
+#define WS_MAGIC "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"
+
+static void ws_mask_payload(u8_t *payload, size_t payload_len,
+			    u32_t masking_value)
+{
+	int i;
+
+	for (i = 0; i < payload_len; i++) {
+		payload[i] ^= masking_value >> (8 * (3 - i % 4));
+	}
+}
+
+void ws_mask_pkt(struct net_pkt *pkt, u32_t masking_value)
+{
+	struct net_buf *frag;
+	int i, count = 0;
```

Comment:
Similar comment to the above (count belongs to websocket_ctx).

Reply:
I lost the context here, what do you mean by this?